examples/example_bids_datasets.py

```diff
@@ -0,0 +1,187 @@
+    ''' To make an event file GLMsingle compatible, calculate total
+    acquisition time rounded to nearest second. Total time will represent
+    number of rows in new event file. '''
+    # Print run data
+    tr = metadata['RepetitionTime']
+    num_slices = slicedata.shape[3]
+    total_dur = round(tr * num_slices)
+    print('TR:', tr, '\n Number of slices acquired:', num_slices,
+          '\n Total run duration:', total_dur)
+    return raw, processed, tr, num_slices, total_dur
+
+def make_new_df(root, raw, subjects, total_dur):
+    ''' Make an empty dataframe, # of rows is equal to total run duration (seconds),
+    because we will upsample the TR to 1s to reduce jitter. # of columns equal to
+    total unique event types in .tsv file. In this case 3 unique event types exist,
+    as we ignore the event type representing intertrial intervals (ITIs). '''
+    # Grab event files
+    eventfiles = raw.get(subject=subjects, extension='.tsv', suffix='events', return_type='filename')
+    # Load into pandas
+    for eventfile in eventfiles:
+        design = pd.read_csv(eventfile, sep='\t')
+        # Parse each event type, as indicated by value, where each integer represents a different event type
+        CSplus_shock = design.loc[design['value'] == 3]
+        CSplus = design.loc[design['value'] == 1]
+        CSminus = design.loc[design['value'] == 2]
+        # Make new df
+        new_design = pd.DataFrame(columns = ['TR', 'CSplusshock','CSplus','CSminus'])
+        new_design['TR'] = [i for i in range(1, (total_dur+1))]
+        # For each stimulus type, mark the onset of the stimulus with 1. All other values should be 0
+        for cs in [CSplus_shock, CSplus, CSminus]:
+            for i in range(0, len(cs.index)):
+                tr_start = int(round(cs.iloc[i,0]))
+                if cs is CSplus_shock:
+                    new_design.iloc[tr_start, :] =  [tr_start, 1, 0, 0]
+                elif cs is CSplus:
+                    new_design.iloc[tr_start, :] =  [tr_start, 0, 1, 0]
+                elif cs is CSminus:
+                    new_design.iloc[tr_start, :] =  [tr_start, 0, 0, 1]
+                else:
+                    new_design.iloc[tr_start, :] =  [tr_start, 0, 0, 0]
+        # Fill NaNs with 0s
+        new_design.fillna(0, inplace=True)
+        # Give the event file a new name
+        new_name = raw.parse_file_entities(eventfile)
+        new_name['suffix'] = 'events_glmsingle'
+        pattern = 'sub-{}_ses-{}_task-{}_run-{}_{}.csv'
+        # Save the event file in derivatives, exclude TR column
+        new_design.to_csv(os.path.join(root, 'derivatives', 'iter4',
+        'sub-'+new_name['subject'], 'ses-'+new_name['session'], 'func',
+        pattern.format(new_name['subject'],
+        new_name['session'], new_name['task'], new_name['run'],
+        new_name['suffix'])), columns=['CSplusshock','CSplus','CSminus'],
+        index=False, header=False)
```

Comment:
You probably :
A) want to use pybids build_path methods (https://bids-standard.github.io/pybids/generated/bids.layout.writing.html#module-bids.layout.writing) that should happily accept the dict of entities you have just created.
B) want to give a BIDS friendly name to your output file: keep `events` as suffix but add a `desc-glmsingle` entity